src/hipscat/catalog/pixel_node.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+from typing import List
+
+from hipscat.catalog.pixel_node_type import PixelNodeType
+
+
+class PixelNode:
+    """A node in the HIPSCat quadtree of HEALPix pixels
+
+    Attributes:
+        hp_order: HEALPix order of the pixel
+        hp_pixel: HEALPix pixel number
+        node_type: If the node is a leaf, root, or inner type
+        parent: The parent pixel node in the tree
+        children: The child nodes of the pixel
+
+    """
+
+    def __init__(self,
+                 hp_order: int | None,
+                 hp_pixel: int | None,
+                 node_type: PixelNodeType,
```

Comment:
Is the type of node always fixed? Or can it ever change? I'm picturing a case where a leaf node gets too large, becomes an internal node, and is split into 4 leaves. I don't know if this class is mean to support that use case or not though.

Reply:
I don't think there's anything currently in this repo to support any editing of catalogs. There's nothing in the class that currently supports a use case like this, but I don't think the design prohibits adding that functionality later (by changing the node_type). But I can work on supporting this now if you think it's necessary.

Reply:
I don't think it is necessary to change it now as long as we aren't locking ourselves into anything.